### PR TITLE
repat only incorrect doesnt work yet

### PR DESCRIPTION
**Pull request by Adam3004**

### CODE REVIEW

File: src/pages/quiz/summmary/QuizPage.tsx [hunk 1]
```diff
@@ -88,7 +92,16 @@ const QuizPage = ({quizId, workspaceId}: { quizId: string, workspaceId: string }
             navigation.navigate("QuestionsScreen", {
                 quizId: quizId,
                 questions: questions,
-                quiz: quiz
+                quiz: quiz,
+                previouslyCorrect: 0
+            });
+        };
+        const navigateToOnlyIncorrectQuestionScreen = () => {
```
Comment (Adam3004) [inline src/pages/quiz/summmary/QuizPage.tsx:101]: it still navigate to the QuestionScreen with all question. I checked it and backend restuns correct numebr of questions but it is not working there

File: src/pages/quiz/summmary/QuizPage.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import AuthorizedResource from "../../AuthorizedResource";
 import QuizDrawer from "../../notes/QuizDrawer";
 import {useAuth} from "../../auth/AuthProvider";
 import {ModularTopBar, OptionsButtons, UserDetailsWithMenu} from "../../../components/topbar";
+import {isDisabled} from "@expo/metro-runtime/build/error-overlay/Data/LogBoxData";
```
Comment (krisx303) [inline src/pages/quiz/summmary/QuizPage.tsx]: ?

File: src/pages/quiz/summmary/QuizPage.tsx [hunk 3]
```diff
@@ -88,7 +92,16 @@ const QuizPage = ({quizId, workspaceId}: { quizId: string, workspaceId: string }
...
+                quiz: quiz,
+                previouslyCorrect: 0
+            });
+        };
+        const navigateToOnlyIncorrectQuestionScreen = () => {
+            navigation.navigate("QuestionsScreen", {
+                quizId: quizId,
+                questions: incorrectQuestions,
+                quiz: quiz,
+                previouslyCorrect: quiz.bestScore.correct
```
Comment (krisx303) [inline src/pages/quiz/summmary/QuizPage.tsx]: bestScore or lastScore?

File: src/pages/quiz/summmary/QuizPage.tsx (hunk 1 above)
Comment (krisx303) [reply src/pages/quiz/summmary/QuizPage.tsx:101]: It seems that values are passed but react native does not rerender the view (it caches views to minimalize the redundant rerendering each time when you swith page), but we can force rerender by adding this simple effect (if route changes then reset stats):
```
    useEffect(() => {
        setCorrectness([]);
        setQuestion(questions[0]);
        setUserAnswer(getBaseUserAnswer(questions[0]));
        setIndex(0);
    }, [questions, quiz, previouslyCorrect]);
```

At least it seems to work for me 😅

File: src/pages/quiz/summmary/QuizPage.tsx [hunk 4]
```diff
@@ -54,6 +55,9 @@ const QuizPage = ({quizId, workspaceId}: { quizId: string, workspaceId: string }
         httpClient.getQuizQuestions(quizId)
             .then(setQuestions)
             .catch(console.error);
+        httpClient.getIncorrectQuizQuestions(quizId)
```
Comment (krisx303) [inline src/pages/quiz/summmary/QuizPage.tsx:57]: Add this also to `useFocusEffect` function to reload incorrect questions each time user enters this view. We need that to disable button after user solves quiz to 100%
I know, react native is weird 👀

**Review by krisx303 — APPROVED** [review]
Generally ok, but fix the view caching issue before merging

File: src/pages/quiz/summmary/QuizPage.tsx (hunk 2 above)
Comment (Adam3004) [reply src/pages/quiz/summmary/QuizPage.tsx]: long story

**Pull request merged**